oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/ReactiveOidcIdTokenDecoderFactory.java

```diff
@@ -239,4 +242,15 @@ public void setClaimTypeConverterFactory(
 		this.claimTypeConverterFactory = claimTypeConverterFactory;
 	}
 
+	/**
+	 * Sets the custom web client that will be used in {@link NimbusReactiveJwtDecoder}.
+	 * The default webClient is created by {@code WebClient.create()}.
+	 * This is optional method if we need to set custom web client in {@link NimbusReactiveJwtDecoder}.
+	 *
+	 * @param webClient webclient
+	 */
+	public void setWebClient(WebClient webClient) {
+		this.webClient = webClient;
```

Comment:
Please check for `null` using `Assert.notNull("webClient cannot be null")`.

Reply:
done